pkg/services/store/k8saccess/dashboard_service.go

```diff
@@ -10,19 +10,26 @@ import (
 )
 
 type k8sDashboardService struct {
-	orig  dashboards.DashboardService
-	store entity.EntityStoreServer
+	orig   dashboards.DashboardService
+	store  entity.EntityStoreServer
+	access K8SAccess
 }
 
 var _ dashboards.DashboardService = (*k8sDashboardService)(nil)
 
-func NewDashboardService(orig dashboards.DashboardService, store entity.EntityStoreServer) dashboards.DashboardService {
+func NewDashboardService(orig dashboards.DashboardService, store entity.EntityStoreServer, access K8SAccess) dashboards.DashboardService {
 	return &k8sDashboardService{
-		orig:  orig,
-		store: store,
+		orig:   orig,
+		store:  store,
+		access: access,
 	}
 }
 
+func (dr *k8sDashboardService) Run(ctx context.Context) error {
```

Comment:
Isn't this redundant given that we have https://github.com/grafana/grafana/blob/fae56b05b410be49acc0704c94af9b3c1d63df46/pkg/services/store/k8saccess/controller.go#L27-L57, and that we eventually want the code from the link to be autogenerated? Is this supposed to be an alternative implementation?

Reply:
I'll dig some more to be sure...